internal/controller/services/servicemesh/servicemesh_controller_actions.go

```diff
@@ -0,0 +1,495 @@
+			),
+		)
+
+		return nil
+	}
+
+	// create authorino namespace if it does not exist
+	authorinoNamespace, err := getAuthorinoNamespace(rr)
+	if err != nil {
+		return fmt.Errorf("failed to obtain Authorino namespace from ServiceMesh CR: %w", err)
+	}
+	if _, err := cluster.CreateNamespace(
+		ctx,
+		rr.Client,
+		authorinoNamespace,
+		cluster.OwnedBy(sm, rr.Client.Scheme()),
+		cluster.WithLabels(labels.ODH.OwnedNamespace, "true"),
+	); err != nil {
+		return fmt.Errorf("failed to create Authorino namespace %s: %w", authorinoNamespace, err)
+	}
+
+	rr.Templates = append(
+		rr.Templates,
+		odhtypes.TemplateInfo{
+			FS:   resourcesFS,
+			Path: authorinoTemplate,
+		},
+		odhtypes.TemplateInfo{
+			FS:   resourcesFS,
+			Path: authorinoServiceMeshMemberTemplate,
+		},
+		odhtypes.TemplateInfo{
+			FS:   resourcesFS,
+			Path: authorinoServiceMeshControlPlaneTemplate,
+		},
+	)
+
+	return nil
+}
+
+func updateMeshRefsConfigMap(ctx context.Context, rr *odhtypes.ReconciliationRequest) error {
+	sm, ok := rr.Instance.(*serviceApi.ServiceMesh)
+	if !ok {
+		return fmt.Errorf("resource instance %v is not a serviceApi.ServiceMesh)", rr.Instance)
+	}
+
+	data := map[string]string{
+		"CONTROL_PLANE_NAME": sm.Spec.ControlPlane.Name,
+		"MESH_NAMESPACE":     sm.Spec.ControlPlane.Namespace,
+	}
+
+	meshRefsConfigMap := &corev1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      meshRefsConfigMapName,
+			Namespace: rr.DSCI.Spec.ApplicationsNamespace,
+		},
+		Data: data,
+	}
+	if err := controllerutil.SetControllerReference(sm, meshRefsConfigMap, rr.Client.Scheme()); err != nil {
+		return fmt.Errorf("error setting owner reference to ConfigMap: %s", meshRefsConfigMapName)
+	}
+
+	if err := rr.AddResources(meshRefsConfigMap); err != nil {
+		return fmt.Errorf("error adding resource (ConfigMap): %s", meshRefsConfigMapName)
+	}
+
+	return nil
+}
+
+func updateAuthRefsConfigMap(ctx context.Context, rr *odhtypes.ReconciliationRequest) error {
+	sm, ok := rr.Instance.(*serviceApi.ServiceMesh)
+	if !ok {
+		return fmt.Errorf("resource instance %v is not a serviceApi.ServiceMesh)", rr.Instance)
+	}
+
+	audiences := sm.Spec.Auth.Audiences
+	audiencesList := ""
+	if audiences != nil && len(*audiences) > 0 {
+		audiencesList = strings.Join(*audiences, ",")
+	}
+
+	authorinoNamespace, err := getAuthorinoNamespace(rr)
+	if err != nil {
+		return fmt.Errorf("failed to obtain Authorino namespace from ServiceMesh CR: %w", err)
+	}
+
+	data := map[string]string{
+		"AUTH_AUDIENCE":   audiencesList,
+		"AUTH_PROVIDER":   authProviderName,
+		"AUTH_NAMESPACE":  authorinoNamespace,
+		"AUTHORINO_LABEL": authorinoLabel,
+	}
+
+	authRefsConfigMap := &corev1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      authRefsConfigMapName,
+			Namespace: rr.DSCI.Spec.ApplicationsNamespace,
+		},
+		Data: data,
+	}
+	if err := controllerutil.SetControllerReference(sm, authRefsConfigMap, rr.Client.Scheme()); err != nil {
```

Comment:
probably not required, the deploy action sets the controller reference automatically

Reply:
same as above